File: Solutions/66. Plus One/66. Plus One.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> res(digits.size()+1);
        for(int i=digits.size()-1;i>=0;--i){
            if(digits[i]<9){
                ++digits[i];
                return digits;
            }
            digits[i]=0;
        }
        res[0]=1;
        return res;
    }
};

File: Solutions/66. Plus One/66. Plus One.go
func plusOne(digits []int) []int {
    res:=make([]int,len(digits)+1)
    for i:=len(digits)-1;i>=0;i-=1{
        if(digits[i]<9){
            digits[i]+=1
            return digits
        }
        digits[i]=0
    }
    res[0]=1
    return res
}